treeherder/webapp/api/performance_serializers.py

```diff
@@ -146,3 +150,49 @@ class IssueTrackerSerializer(serializers.ModelSerializer):
     class Meta:
         model = IssueTracker
         fields = ['id', 'text', 'issueTrackerUrl']
+
+
+class PerformanceQueryParamsSerializer(serializers.Serializer):
+    startday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    endday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    revision = serializers.CharField(required=False, allow_null=True, default=None)
+    repository = serializers.CharField()
+    framework = serializers.ListField(child=serializers.IntegerField())
+    interval = serializers.IntegerField(required=False, allow_null=True, default=None)
+    parent_signature = serializers.CharField(required=False, allow_null=True, default=None)
+    no_subtests = serializers.BooleanField(required=False)
+
+    def validate(self, data):
+        if data['revision'] is None and (data['startday'] is None or data['endday'] is None):
+            raise serializers.ValidationError('Required: revision or startday and endday.')
+
+        return data
+
+    def validate_repository(self, repository):
```

Comment:
this method doesn't appear to be used

Reply:
It's used by this serializer to validate the query params (it'll validate the repository entered and either return an error message or not, if its valid). You can see it in action [here](https://treeherder-prototype.herokuapp.com/api/perfbyrevision/?framework=1&interval=172800&subtests=0&repository=mozillacentral&revision=413020). It'll also be integrated into the UI error handling during the switch to React.